### PR TITLE
Add timeout to avoid hang reading stats from NSQ

### PR DESCRIPTION
**Pull request by timonwong**

### CODE REVIEW

File: collector/stats.go [hunk 1]
```diff
@@ -84,9 +86,33 @@ func getPercentile(t *topic, percentile int) float64 {
...
+	return &statsClient{
+		nsqdURL: nsqdURL,
+		httpClient: &http.Client{
+			Transport: &http.Transport{
+				Dial: func(network, addr string) (net.Conn, error) {
+					c, err := net.DialTimeout(network, addr, timeout)
+					if err != nil {
+						return nil, err
+					}
+					if err := c.SetDeadline(time.Now().Add(timeout)); err != nil {
```
Comment (nordicdyno) [inline collector/stats.go]: In worst case canceled request time could be close to timeout*2 (dial + read)
Please consider using http.Client.Timeout instead.

Useful reference about Go's server/client timeouts: https://blog.cloudflare.com/the-complete-guide-to-golang-net-http-timeouts/

File: collector/stats.go [hunk 2]
```diff
@@ -84,9 +86,33 @@ func getPercentile(t *topic, percentile int) float64 {
 	return 0
 }
 
-func getNsqdStats(nsqdURL string) (*stats, error) {
+type statsClient struct {
```
Comment (nordicdyno) [inline collector/stats.go]: I'm not sure we need additional type here just for fetching

File: collector/stats.go (hunk 1 above)
Comment (edganiukov) [reply collector/stats.go]: 👍  I would suggest to set Client.Timeout instead

**Comment by nordicdyno** [comment]
LGTM

**Review by heltonmarx — APPROVED** [review]

**Pull request closed without merging**